streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -114,6 +116,15 @@ public void onPartitionsRevoked(final Collection<TopicPartition> revokedPartitio
             try {
                 // suspend only the active tasks, reassigned standby tasks will be closed in onPartitionsAssigned
                 suspendedTasks = taskManager.suspendActiveTasksAndState(revokedPartitions);
+                if (streamThread.eosBetaEnabled && !streamThread.eosUpgradeModeEnabled) {
```

Comment:
Similarly here, when suspending tasks we can just have this logic inside stream-thread and stream-task.